lib/index.js

```diff
@@ -67,16 +71,33 @@ module.exports = {
 };
 
 
-internals.decorate = function (settings, cache) {
+internals.decorateRequest = function (settings, cache) {
 
     return function (request) {
 
-        return new internals.Yar(request, settings, cache);
+        return new internals.RequestYar(request, settings, cache);
     };
 };
 
+internals.decorateServer = function (cache) {
+
+    return new internals.ServerYar(cache);
+};
+
+internals.ServerYar = class {
```

Comment:
Can you change those to be `class ServerYar` and `RequestYar` respectively?
Anonymous classes can be annoying to debug :)